### PR TITLE
Issue-53. Multiple field support for partitioning.

### PR DESCRIPTION
**Pull request by jcustenborder**

### CODE REVIEW

**Comment by jcustenborder** [comment]
[clabot:check]

File: hive/src/main/java/io/confluent/connect/storage/hive/schema/DefaultSchemaGenerator.java [hunk 1]
```diff
@@ -36,9 +36,16 @@ public DefaultSchemaGenerator(Map<String, Object> config) {
   }
 
   @Override
-  public List<FieldSchema> newPartitionFields(String partitionField) {
-    return Collections.singletonList(
-        new FieldSchema(partitionField, TypeInfoFactory.stringTypeInfo.toString(), "")
-    );
+  public List<FieldSchema> newPartitionFields(String format) {
+    String[] fields = format.split(",");
+    final List<FieldSchema> result = new ArrayList<>();
```
Comment (kkonstantine) [inline hive/src/main/java/io/confluent/connect/storage/hive/schema/DefaultSchemaGenerator.java]: nit: Access pattern of result doesn't differ from that of `fields`. They are both assigned once. You may remove `final` here to make this clear.

File: partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldPartitioner.java [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 package io.confluent.connect.storage.partitioner;
 
+import io.confluent.connect.storage.common.StorageCommonConfig;
```
Comment (kkonstantine) [inline partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldPartitioner.java]: nit: noise due to code style. You might want to skip these changes to reduce the chance of future merge conflicts.

File: partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldPartitioner.java [hunk 3]
```diff
@@ -72,7 +84,14 @@ public String encodePartition(SinkRecord sinkRecord) {
...
   public List<T> partitionFields() {
     if (partitionFields == null) {
-      partitionFields = newSchemaGenerator(config).newPartitionFields(fieldName);
+      StringBuilder builder = new StringBuilder();
+      for (String f : fieldNames) {
+        if (builder.length() > 0) {
+          builder.append(',');
+        }
+        builder.append(f);
+      }
```
Comment (cricket007) [inline partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldPartitioner.java]: Could use `org.apache.kafka.common.utils.Utils.join` here

**Comment by cricket007** [comment]
Doesn't this duplicate #30 ?

**Comment by jcustenborder** [comment]
#30 has been abandoned by the developer and does not merge.

**Comment by kkonstantine** [comment]
Looks great @jcustenborder! Thanks for taking a stab at this. Merging

**Pull request merged**